### DISCUSSION TITLE
Upload multiple files and folders

### DISCUSSION THREAD

**Discussion by satishbkodali** [octokit/octokit.js#1766] [Q&A] [answered]
currently I see only this api function to update files

```js
octokit.repos.createOrUpdateFileContents({
  owner, repo, path, message, content, committer, author
})
```

But will this update or create recursive folders and with multiple files in those folders too?

**Comment by gr2m** [comment, marked as answer]
No, the `octokit.repos.createOrUpdateFileContents` method will only create or update a single file at the time. If you create a file in a folder that does not exist yet, the folders will be created automatically.

In order to create/update multiple files, you an call the `octokit.repos.createOrUpdateFileContents` method multiple times. If you want to edit multiple files in a single commit, you will need to use the `octokit.git.*` APIs. If you are interested, you can use my [Octokit plugin to create a pull request](https://github.com/gr2m/octokit-create-pull-request) as a reference.
